### ISSUE TITLE
Add griffpatch's physics engine

### ISSUE THREAD

**Issue by BonfireScratch** [LLK/scratch-gui#4402]
### Expected Behavior

Griffpatch' s physics engine must be an extension

### Actual Behavior

It doesn't exist

### Steps to Reproduce

None

### Operating System and Browser

Any

**Comment by apple502j** [comment]
Disagree - if someone claimed that "My physics engine should be added too!", what will you do?

**Comment by BonfireScratch** [comment]
@apple502j that's a good point. However, the ST may not say who made the engine. Of course, if

**Comment by griffpatch** [comment]
Hey now... I haven't even committed it to github, let alone got it ready
for any kind of release. Extensions will most likely not be considered for
quite some time to come. What we can do is try to host our extensions on
GitHub so that you can play with the ideas in a safe environment outside of
scratch... If I can work it out :/ keep struggling with getting git to do
what I want!

**Comment by ericrosenbaum** [comment]
I love @griffpatch's physics experiments!

I did some experiments along these lines, before we even had an extension spec. I used matter.js, but the idea was pretty similar. We may not be able to include something like this as an official extension, for various reasons, but it's worth thinking about. Here are some demos I made back in summer 2017, I think inspired by earlier work from griffpatch (note that the first one doesn't even have any blocks!):

![physics-demo1](https://user-images.githubusercontent.com/567844/51562469-e2594500-1e57-11e9-9f11-095618749d66.gif)

![physics-billiards](https://user-images.githubusercontent.com/567844/51562465-e08f8180-1e57-11e9-82f9-2e7d73cd2650.gif)

![angry-bus](https://user-images.githubusercontent.com/567844/51562457-dc636400-1e57-11e9-9deb-d9ca4e3f907f.gif)

**Comment by benjiwheeler** [comment]
I think this is a very intriguing idea. I've often wished for an easier way to add simple 2d gravity and velocity setting, e.g. for platformers, since that's something many of my students immediately show interest in. 

 I can see the argument against it -- that someone trying to, say, make their first platformer would learn more starting with simple constant speed up, constant speed down and then iterating from there, rather than using an existing library. But, an extension would not replace the ability to learn this.

**Comment by griffpatch** [comment]
Interesting! My first attempt at the scratch 3 physics extension also used
matter.is, but after a little playing with it i found it not very good...
Box2d provided far better physics interactions and allow for the required
offset of the sprite center vs. their center of mass... :)

I'm having a bit of trouble with my GitHub branches... I want to revert my
branch to a fresh new copy of your repo and then upload to a working GitHub
site... But not am I struggling with it *sigh*. Feel like it should be so
easy ;)

**Comment by griffpatch** [comment]
In case you're interested, here's a video of the current extension in
action:

https://youtu.be/HRa9QcCQMHU

**Comment by BonfireScratch** [comment]
@benjiwheeler Or we could use box2d.js as an alternative... But I don't recommend that

**Comment by BonfireScratch** [comment]
@apple502j I don't think it would happen, especially if people don't know who made it.

**Comment by benjiwheeler** [comment]
@griffpatch I can try to help with Github stuff... feel free to reach out more directly, my info is at techno-social.com

**Comment by griffpatch** [comment]
> @griffpatch I can try to help with Github stuff... feel free to reach out more directly, my info is at techno-social.com

Thanks Benji, but as luck would have it I think I finally got my repo in order and managed to publish my scratch-gui to github pages: https://griffpatch.github.io/scratch-gui/ Ta-da!

**Comment by alexwenbj** [comment]
@griffpatch It is an amazing and useful extension for Scratch, thank you for bringing it for scratch.
I cloned your scratch-gui repo(https://github.com/griffpatch/scratch-gui, develp branch) and built it,but when I opened via http://0.0.0.0:8601/,and intended to open the box2d extension,but it failed and got error msg: `DOMException: Failed to execute 'importScripts' on 'WorkerGlobalScope': The script at 'http://0.0.0.0:8601/griffpatch' failed to load.`, it seems the Extension is not included in the repo. How and where I can get it? Thank you again.
